resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -59,7 +61,24 @@ object ExecutorPodsSnapshot extends Logging {
         case "pending" =>
           PodPending(pod)
         case "running" =>
-          PodRunning(pod)
+          // TODO: upcoming Kubernenetes feature will make this code redundant
```

Comment:
Hi, @jinxingwang .
- First of all, in the shared repository like Apache Spark, we had better use **IDed** `TODO`. If this is needed, you had better use `TODO(SPARK-XXXXX)` style.
- Second, please use the exact version instead of `upcoming` or `latest` because those words become obsolete very soon especially in K8s domain.
- According to the PR content, it looks like you are saying we can skip this PR technically. Could you be more clear why we need to have this code?

Reply:
We needed because currently there is no solution from k8s yet, and this is a bug.
Also even there is a solution, that will require the users to upgrade k8s.

Reply:
Then, please enumerate the exact K8s versions in the PR description which you target in this PR. Also, you need to create another JIRA issue to remove this TODO (and your code) later.

Reply:
Will do, thanks!